java/vector/src/main/java/org/apache/arrow/vector/NullVector.java

```diff
@@ -0,0 +1,287 @@
+    if (!children.isEmpty()) {
+      throw new IllegalArgumentException("Null vector has no children");
+    }
+  }
+
+  @Override
+  public List<FieldVector> getChildrenFromFields() {
+    return Collections.emptyList();
+  }
+
+  @Override
+  public void loadFieldBuffers(ArrowFieldNode fieldNode, List<ArrowBuf> ownBuffers) {
+    if (!ownBuffers.isEmpty()) {
+      throw new IllegalArgumentException("Null vector has no buffers");
+    }
+  }
+
+  @Override
+  public List<ArrowBuf> getFieldBuffers() {
+    return Collections.emptyList();
+  }
+
+  @Override
+  public List<BufferBacked> getFieldInnerVectors() {
+    return Collections.emptyList();
+  }
+
+  @Override
+  public long getValidityBufferAddress() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public long getDataBufferAddress() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public long getOffsetBufferAddress() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public ArrowBuf getValidityBuffer() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public ArrowBuf getDataBuffer() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public ArrowBuf getOffsetBuffer() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public int getValueCount() {
+    return this.valueCount;
+  }
+
+  @Override
+  public void setValueCount(int valueCount) {
+    this.valueCount = valueCount;
+  }
+
+  @Override
+  public Object getObject(int index) {
+    return null;
+  }
+
+  @Override
+  public int getNullCount() {
+    return this.valueCount;
+  }
+
+  @Override
+  public boolean isNull(int index) {
+    if (index < valueCount) {
+      return true;
+    }
+    throw new IndexOutOfBoundsException();
+  }
+
+  @Override
+  public int hashCode(int index) {
+    return 0;
+  }
+
+  @Override
+  public <OUT, IN> OUT accept(VectorVisitor<OUT, IN> visitor, IN value) {
+    return visitor.visit(this, value);
+  }
+
+  @Override
+  public void copyFrom(int fromIndex, int thisIndex, ValueVector from) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void copyFromSafe(int fromIndex, int thisIndex, ValueVector from) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public String getName() {
+    return this.getField().getName();
+  }
+
+  private class TransferImpl implements TransferPair {
+    NullVector to;
+
+    public TransferImpl() {
+      to = new NullVector();
+    }
+
+    public TransferImpl(NullVector to) {
+      this.to = to;
+    }
+
+    @Override
+    public NullVector getTo() {
+      return to;
+    }
+
+    @Override
+    public void transfer() {
+      to.valueCount = valueCount;
+    }
+
+    @Override
+    public void splitAndTransfer(int startIndex, int length) {
+      to.valueCount = length;
+    }
+
+    @Override
+    public void copyValueSafe(int fromIndex, int toIndex) {}
```

Comment:
I think this needs to set value count on the target if toIndex is larger then the current index?

Reply:
fixed.